### PR TITLE
Add force flag to init new site inside non empty dir

### PR DESCRIPTION
**Pull request by ns3777k**
Add force flag

If flag is passed the site will be created inside non-empty folder
only if there are no existent folders or config with the same name
hugo creates.

Resolves: https://github.com/spf13/hugo/issues/1163

### CODE REVIEW

**Comment by bep** [comment]
What do you mean by "x,y,z"?

Also, could you get your commit message in line with the guidelines?

File: commands/new.go [hunk 1]
```diff
@@ -117,13 +119,24 @@ func NewSite(cmd *cobra.Command, args []string) {
...
+		if isDir, _ := helpers.IsDir(createpath, hugofs.SourceFs); !isDir {
+			jww.FATAL.Fatalln(createpath, "already exists but not a directory")
+		}
+
+		if isEmpty, _ := helpers.IsEmpty(createpath, hugofs.SourceFs); !isEmpty && !forceNew {
 			jww.FATAL.Fatalln(createpath, "already exists and is not empty")
 		}
+
+		if forceNew {
+			if err = os.RemoveAll(createpath); err != nil {
```
Comment (bep) [inline commands/new.go]: Remove all? No, this isn't what the user would expect.

Comment (ns3777k) [inline commands/new.go]: Only if you pass *force* flag. Can make a make up

Comment (bep) [inline commands/new.go]: The flag help text says "create even if destination is not empty"". It says nothing about deleting stuff. And rightfully so.

Comment (ns3777k) [inline commands/new.go]: I edited flag help text & commit message.
The variables inside the method were called like "x", "y" and "z".
I renamed them to increase readability.

**Comment by anthonyfok** [comment]
Hi @ns3777k,

I think even with the `--force` flag, it should not remove _any_ files or directories.
If it needs to overwrite any existing files, e.g. an existing `config.toml`, it should either abort or prompt the end user what to do.

Otherwise, this `--force` that you currently have is a "killer"---no, an "assassin" command that may seem innocuous to the end user but end up having his/her entire home directory deleted when he/she mistakenly type `hugo --force new site .`

**Comment by bep** [comment]
Agree with @anthonyfok No deletions. Not even after prompting the user. I do not want to be responsible for wiping out peoples stuff.

**Comment by ns3777k** [comment]
Misunderstood the issue. Got you. Get done on Sunday.

**Comment by ns3777k** [comment]
Edited so there's no prompt or deletion.

**Comment by spf13** [comment]
@ns3777k Thanks for this. The users will really benefit from it. It would be great to have a test accompanying it since it has gotten a bit complex.

**Comment by ns3777k** [comment]
@spf13 Done.
Not sure If I chose the right way to make it testable so any comments are welcome :-)

**Comment by spf13** [comment]
@ns3777k Please click on the "Details" link to the right of the license/cla line and click "approve" so we can merge this. The CLA isn't new, but this integration is. 

Thanks!

**Comment by ns3777k** [comment]
@spf13 Approved. Still pending here though.

**Comment by bep** [comment]
Merged in 5e97cf3020284859ddaba528801329a54cf95847

**Pull request closed without merging**